pynitrokey/cli/storage.py

```diff
@@ -30,24 +40,117 @@ def list():
 
 
 @click.command()
-@click.option(
-    "-p",
-    "--password",
-    default="12345678",
-    help="update password to be used instead of default",
-)
-def enable_update(password):
-    """enable firmware update for NK Storage device"""
+def enable_update():
+    """enable firmware update for NK Storage device
 
+    If the Admin Pin is not in the environment variable NITROPY_ADMIN_PIN, it will be prompted from stdin
+    """
+    password = AskUser("Admin Pin", envvar="NITROPY_ADMIN_PIN", hide_input=True).ask()
     local_print("Enabling firmware update mode")
-    nks = NitrokeyStorage()
-    nks.connect()
-    try:
-        if nks.enable_firmware_update(password) == 0:
-            local_print("setting firmware update mode - success!")
-    except DeviceNotFound:
-        local_print("No Nitrokey Storage device found")
+    nks = connect_nkstorage()
+    if nks.enable_firmware_update(password) == 0:
+        local_print("setting firmware update mode - success!")
```

Comment:
I missed that earlier, sorry. 
1. NK Storage has 3 different passwords - smart card related User PIN and Admin PIN, and MCU related (firmware) update password. The latter should be used here, not the Admin PIN. 
2. Misspelling of PIN (is `Pin`) - please check in other places as well.

Reply:
That could have been bad. Fixed it, the env variable is now `NITROPY_FIRMWARE_PASSWORD`